opal/util/ethtool.c

```diff
@@ -0,0 +1,66 @@
+#include "opal_config.h"
+
+#include <string.h>
+#include <limits.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+#ifdef HAVE_NET_IF_H
+#include <net/if.h>
+#endif
+#ifdef HAVE_LINUX_ETHTOOL_H
+#include <linux/ethtool.h>
+#endif
+#ifdef HAVE_SYS_IOCTL_H
+#include <sys/ioctl.h>
+#endif
+#ifdef HAVE_LINUX_SOCKIOS_H
+#include <linux/sockios.h>
+#endif
+
+#include "opal/util/ethtool.h"
+#include "opal/util/if.h"
+
+/*
+ * Set an appropriate bandwidth for the btl instance. On Linux, we obtain this
+ * via an ioctl(). Elsewhere or in the error case, we return the speed as 0.
+ */
+unsigned int
+opal_ethtool_get_speed (const char *if_name)
+{
+    unsigned int speed = 0;
+
+#if !defined(HAVE_DECL_SIOCETHTOOL) || !defined(HAVE_STRUCT_IFREQ) || !defined(HAVE_STRUCT_ETHTOOL_CMD)
```

Comment:
I think you also need to also check for ethtool_cmd_speed (you check for it in configure).